src/validators/auth.ts

```diff
@@ -112,3 +113,46 @@ export async function validateRefreshToken(
     }
   }
 }
+
+/**
+ * check for existing user
+ *
+ * @param  {Request}   req
+ * @param  {Response}   res
+ * @param  {NextFunction} next
+ * @returns {Promise}
+ */
+export async function checkUser(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const mobile = req.body.mobile;
+    const email = req.body.email;
+
+    const [userMobile] = await userService.search({ mobile });
+    const [userEmail] = await userService.search({ email });
+
+    if (!userMobile && !userEmail) {
+      next();
+    } else {
+      let errorString = '';
+
+      if (userMobile) {
+        errorString += `${mobile} `;
+      }
+      if (userMobile && userEmail) {
+        errorString += `& ${email} `;
+      } else if (userEmail) {
+        errorString += `${email} `;
+      }
+
+      errorString += 'already exists';
```

Comment:
We shouldn't do this as a string. We should rather create an object with key = mobile and error text as value.
Same with the email part. This should be done this way to give the front-end the feasibility to identify the problem rather than just displaying the problem as a text.

```suggestion

let errorPayload = ();

if(userMobile) errorPayload.mobile = "Mobile Already Exists.";
if(userEmail) errorPayload.email = "Email Already Exists.";


```